app/models/course.rb

```diff
@@ -98,4 +110,23 @@ def export_students_to_csv
       end
     end
   end
+
+  def delete_roster_students(spreadsheet, header_row_exists, perm_index)
+    perms = sort_perms_into_hash(spreadsheet, header_row_exists, perm_index)
+    self.roster_students.each do |student|
+      perm = perms[student.perm]
+      if perm.blank?
+        student.destroy
+      end
+    end
+  end
+
+  def sort_perms_into_hash(spreadsheet, header_row_exists, perm_index)
```

Comment:
both of these methods should probably be moved into a private section if they aren't already.